### PR TITLE
fix(generator): emitting unnecessary includes and code with no lro or pagination

### PR DESCRIPTION
**Pull request by scotthart**
These changes to the generator correct issues discovered in the code review for the generated IAMCredentials service.
There were some unnecessary includes that were removed, and some added due to transitive includes removed.
Changes were made to emitted code when the service did not have longrunning and/or paginated methods.
Non-method specific `PredicateFragment` print functions were added to facilitate this.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5644)

### CODE REVIEW

File: generator/internal/idempotency_policy_generator.cc [hunk 1]
```diff
@@ -117,11 +117,12 @@ Status IdempotencyPolicyGenerator::GenerateCc() {
   CcPrint(  // clang-format off
     "// Generated by the Codegen C++ plugin.\n"
     "// If you make any local changes, they will be lost.\n"
-    "// source: $proto_file_name$\n\n");
+    "// source: $proto_file_name$\n");
```
Comment (devbww) [inline generator/internal/idempotency_policy_generator.cc:120]: I'd vote to maintain that separator line.

File: generator/internal/logging_decorator_generator.cc [hunk 2]
```diff
@@ -122,13 +122,13 @@ Status LoggingDecoratorGenerator::GenerateCc() {
   CcPrint(  // clang-format off
     "// Generated by the Codegen C++ plugin.\n"
     "// If you make any local changes, they will be lost.\n"
-    "// source: $proto_file_name$\n\n");
+    "// source: $proto_file_name$\n");
```
Comment (devbww) [inline generator/internal/logging_decorator_generator.cc:125]: Ditto.

File: generator/internal/metadata_decorator_generator.cc [hunk 3]
```diff
@@ -123,39 +121,25 @@ Status MetadataDecoratorGenerator::GenerateCc() {
   CcPrint(  // clang-format off
             "// Generated by the Codegen C++ plugin.\n"
             "// If you make any local changes, they will be lost.\n"
-            "// source: $proto_file_name$\n\n");
+            "// source: $proto_file_name$\n");
```
Comment (devbww) [inline generator/internal/metadata_decorator_generator.cc:124]: Ditto.

**Review by devbww — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5644)* status: 0 of 21 files reviewed, 3 unresolved discussions (waiting on @devbww and @scotthart)

___
*[generator/internal/idempotency_policy_generator.cc, line 120 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5644#-MOhdO72-hoo-zncl1se-r1-120:-MOiVbbVBmVPX54MRBcR:b-d4gq7f) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/00a90e594a2a3fdc0611d7bce1100dd30ee8dc66/generator/internal/idempotency_policy_generator.cc#L120)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

I'd vote to maintain that separator line.
</blockquote></details>

Currently it's inconsistent across generated files.  Filed issue #5648 to follow up on a course of action to be decided by the team.

**Pull request merged**